chia/pools/pool_wallet.py

```diff
@@ -272,11 +271,8 @@ async def apply_state_transitions(self, block_spends: List[CoinSpend], block_hei
         The DB must be committed after calling this method. All validation should be done here.
         """
         coin_name_to_spend: Dict[bytes32, CoinSpend] = {cs.coin.name(): cs for cs in block_spends}
-
         tip: Tuple[uint32, CoinSpend] = await self.get_tip()
-        tip_height = tip[0]
         tip_spend = tip[1]
-        assert block_height >= tip_height  # We should not have a spend with a lesser block height
```

Comment:
Why are we removing this? The block height must always be greater than the tip. We cannot apply a state transition that occurred in the past, they must be all applied in the correct order.

Reply:
I changed how the pools wallet works, it will only apply a single state transition at once. new state will have to be a spend of a current tip the pool wallet is tracking.